test/jellyfish_web/socket_test.exs

```diff
@@ -0,0 +1,135 @@
+      assert {:ok, state} = connect(%{"room_id" => room_id, "peer_id" => peer_id})
+      assert state.peer_id == peer_id
+      assert state.room_id == room_id
+
+      status =
+        room_pid
+        |> Room.get_state()
+        |> Map.get(:peers)
+        |> Map.get(peer_id)
+        |> Map.get(:status)
+
+      assert status == :connected
+    end
+
+    test "when peer is already connected", %{room_id: room_id, peer_id: peer_id} do
+      assert {:ok, _state} = connect(%{"room_id" => room_id, "peer_id" => peer_id})
+      assert {:error, :already_connected} = connect(%{"room_id" => room_id, "peer_id" => peer_id})
+    end
+
+    test "when room does not exist" do
+      assert {:error, :room_not_found} =
+               connect(%{"room_id" => "fake_room_id", "peer_id" => "fake_peer_id"})
+    end
+
+    test "when peer does not exist", %{room_id: room_id} do
+      assert {:error, :peer_not_found} =
+               connect(%{"room_id" => room_id, "peer_id" => "fake_peer_id"})
+    end
+
+    test "when request params are missing" do
+      assert {:error, :no_params} = connect(%{"room_id" => "room_id", "not_peer_id" => "abc"})
+    end
+  end
+
+  describe "receiving messages from client" do
+    test "when message is valid Media Event", %{
+      room_id: room_id,
+      peer_id: peer_id,
+      room_pid: room_pid
+    } do
+      :erlang.trace(room_pid, true, [:receive])
+
+      json = Jason.encode!(%{"type" => "mediaEvent", "data" => @data})
+      send_from_client(json, room_id, peer_id)
+
+      # check if room process received the media event
+      assert_receive {:trace, ^room_pid, :receive, {:media_event, ^peer_id, _data}}
+    end
+
+    test "when room does not exist" do
+      json = Jason.encode!(%{"type" => "mediaEvent", "data" => @data})
+
+      assert capture_log(fn -> send_from_client(json, "fake_room_id", "fake_peer_id") end) =~
+               "Trying to send Media Event to room"
+    end
+
+    test "when message is not a json", %{room_id: room_id, peer_id: peer_id} do
+      assert capture_log(fn -> send_from_client("notajson", room_id, peer_id) end) =~
+               "Failed to decode message"
+    end
+
+    test "when message type is unexpected", %{room_id: room_id, peer_id: peer_id} do
+      json = Jason.encode!(%{"type" => 34})
+
+      assert capture_log(fn -> send_from_client(json, room_id, peer_id) end) =~
+               "Received message with unexpected type"
+    end
+
+    test "when message has invalid structure", %{room_id: room_id, peer_id: peer_id} do
+      json = Jason.encode!(%{"notatype" => 45})
+
+      assert capture_log(fn -> send_from_client(json, room_id, peer_id) end) =~
+               "Received message with invalid structure"
+    end
+  end
+
+  describe "receiving messages from server" do
+    test "when it is a valid Media Event" do
+      assert {:push, {:text, json}, _state} = Socket.handle_info({:media_event, @data}, %{})
```

Comment:
nitpick but maybe we could have send_from_server (to be consistent with send_from_client) ?